### PR TITLE
Fix which prints 

### PR DESCRIPTION
**Pull request by melund**
So I am just opening  a new PR for the `which` stuff, which wasn't addressed in @scopatz's #1803 PR

This fixes the print statements when `which` doesn't find any matches, and ensures they go to stderr.

### CODE REVIEW

File: xonsh/xoreutils/which.py [hunk 1]
```diff
@@ -97,7 +97,10 @@ def which(args, stdin=None, stdout=None, stderr=None, spec=None):
 
     pargs = parser.parse_args(args)
     verbose = pargs.verbose or pargs.all
-    captured = spec.captured in xproc.STDOUT_CAPTURE_KINDS
+    if spec is not None:
+        captured = spec.captured in xproc.STDOUT_CAPTURE_KINDS
+    else:
+        captured = False
```
Comment (scopatz) [inline xonsh/xoreutils/which.py]: I think these do the same thing since None is not in `xproc.STDOUT_CAPTURE_KINDS`, right?

**Comment by scopatz** [comment]
Thanks for this @melund!  Just the one small comment, but otherwise, I think is this good to go.

File: xonsh/xoreutils/which.py (hunk 1 above)
Comment (melund) [inline xonsh/xoreutils/which.py]: Exactly. The `spec` could be None if not passed as a kwarg.

Comment (scopatz) [inline xonsh/xoreutils/which.py]: whoops! my bad

**Comment by scopatz** [comment]
Thanks @melund

**Pull request merged**